Add MP61 RepeatActions

tevans78 opened the pull request:
Either add a repeat for MicroProfile 6.1 or replace the existing 6.0 repeat.

fixes https://github.com/OpenLiberty/open-liberty/issues/25913

tevans78 committed:
Allow RepeatActions to work with multiple servers

tevans78 committed:
Add MP61 Repeats to Concurrency FATs

tevans78 committed:
Add MP61 Repeats to CDI FATs

tevans78 committed:
Add MP61 Repeats to GRPC FATs

tevans78 committed:
Add MP61 Repeats to MP Config FATs

tevans78 committed:
Add MP61 Repeats to MP FaultTolerance FATs

tevans78 committed:
Add MP61 Repeats to MP OpenAPI FATs

tevans78 committed:
Add MP61 Repeats to MP Rest Client FATs

tevans78 committed:
Add MP61 Repeats to Request Timing FATs

tevans78 committed:
Add MP61 Repeats to Rest Handler FATs

tevans78 committed:
Add MP61 Repeats to CheckPoint FATs

tevans78 committed:
Add MP61 Repeats to MP Graph QL FATs

tevans78 committed:
Add MP61 Repeats to MP Health FATs

tevans78 committed:
Update Overall MicroProfile FATs

tevans78 commented:
#build

tjwatson commented:
I started an InstantOn build to run the checkpoint tests: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_F5H1MEjGEe6Dy7HKT-o-zA

njr-11 commented on dev/com.ibm.ws.concurrent.mp_fat_jakarta/fat/src/com/ibm/ws/concurrent/mp/fat/jakarta/MPContextProp1_3_EE10_Test.java:
```diff
@@ -58,120 +66,120 @@ public static void tearDown() throws Exception {
+//    @Test
+//    public void testIntermixMicroProfileAndJakartaEECompletionStages() throws Exception {
+//        runTest(server, APP_NAME + "/MPContextProp1_3_EE10_TestServlet", testName);
+//    }
```
Don't leave commented out code/tests around with a good written explanation of why.  It will leave future developers wondering whether it really ought to be that way and whether tests ought to be switched back on.

tevans78 replied:
Sorry, that was an oversight, will remove.

njr-11 commented on dev/com.ibm.ws.concurrent.mp_fat_jakarta/build.gradle:
```diff
@@ -1,17 +1,15 @@
  *******************************************************************************/
 
 addRequiredLibraries.dependsOn addDerby
+addRequiredLibraries.dependsOn addJakartaTransformer
```
This seems odd for a bucket that is already written for jakarta only.

tevans78 replied:
When using JakartaEE9Action or JakartaEE10Action, the transformer code still runs even if there is nothing to transform. No doubt that could be improved but it is a separate issue.

njr-11 commented on dev/com.ibm.ws.concurrent.mp_fat_jakarta/fat/src/com/ibm/ws/concurrent/mp/fat/jakarta/MPContextProp1_3_EE10_Test.java:
```diff
@@ -34,8 +38,12 @@ public class MPContextProp1_3_EE10_Test extends FATServletClient {
     private static final String APP_NAME = "MPContextProp1_3_EE10_App";
 
     @Server("com.ibm.ws.concurrent.mp.fat.1.3.ee10")
+    @TestServlet(servlet = MPContextProp1_3_EE10_TestServlet.class, contextRoot = APP_NAME)
```
Typically, when we write tests that avoid `@TestServlet` and put the `@Test` annotation on the test client side, it's because loading the servlet makes the test bucket fail when run on older Java levels due to something the servlet is compiled against that can't be loaded when the test is making the decision to skip.  Maybe that wasn't the reason here, but you should check what happens on older java levels to be sure.

tevans78 replied:
Interesting, I've not encountered that before. Will check into it.

tevans78 replied:
Tried to fix the FAT framework so that doesn't happen ... but it got messy so I pulled these changes out again.

tevans78 commented:
#build

tjwatson committed:
Add required features to tested.features for checkpoint FAT

tjwatson committed:
Force overwrite of test apps to ensure proper transformation

tjwatson committed:
Disable running FacesBeansTest on EE 10 (temporary)

Some additional work is needed to this test application
to allow it to run on EE10, like removnig usages of
ManagedBean annotations.

This work will be tracked in a separate PR

tjwatson reviewed (changes requested):
For the `io.openliberty.checkpoint_fat` I have three additional commits needed for the tests to pass in my branch at https://github.com/tjwatson/open-liberty/commits/addMP61Repeats2-checkpoint-fixes

tevans78 commented:
#build

Azquelt reviewed (approved):
Approving changes for CDI, Fault Tolerance and OpenAPI

jim-krueger reviewed (approved):
Looks good.   Thankls

tevans78 commented:
#libby
